### PR TITLE
Add ActionBar

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr implements an ActionBar on the Decks- and GroupsTab.

### CODE REVIEW

File: frontend/components/navigation/Tabs/DecksTab.tsx [hunk 1]
```diff
@@ -33,21 +39,31 @@ export const DecksTab = ({ group }: DecksTabProps) => {
...
+										deck.deckName
+											.toUpperCase()
+											.includes(filter)
+									)
+									.reverse()
+							: decks?.filter((deck) =>
+									deck.deckName.toUpperCase().includes(filter)
+							  )
+					}
+				/>
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/DecksTab.tsx:66]: Maybe you could extract the filtering part of the decks (`decks?.filter((deck) => deck.deckName.toUpperCase().includes(filter))`) to 
1. avoid code duplication
2. Resolve this illegal auto-formatting xD

File: frontend/components/navigation/Tabs/GroupsTab.tsx [hunk 2]
```diff
@@ -28,66 +24,49 @@ interface GroupsTabProps {
...
+									)
+									.reverse()
+							: groups.filter(
+									(group) =>
+										!group.isDefault &&
+										group.groupName
+											.toUpperCase()
+											.includes(filter)
+							  )
+					}
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/GroupsTab.tsx:67]: Same here. Also, was it indented that the false block of the ternary does not include the optional chaining operator for groups?

File: frontend/locales/de/messages.po [hunk 3]
```diff
@@ -219,6 +224,14 @@ msgstr "Stapel Beschreibung eingeben"
...
 msgstr "Stapel Namen eingeben"
 
+#: components/modal/CreateGroupModal.tsx:52
+msgid "Enter group description"
+msgstr "Gruppen Beschreibung eingeben"
+
+#: components/modal/CreateGroupModal.tsx:41
+msgid "Enter group name"
+msgstr "Gruppen Namen eingeben"
+
```
Comment (4KevR) [inline frontend/locales/de/messages.po:234]: I think I would rather go with `Gruppenbeschreibung` and `Gruppennamen` but up for discussion.

File: frontend/locales/de/messages.po [hunk 4]
```diff
@@ -249,17 +262,22 @@ msgstr "von"
...
+#: components/modal/CreateGroupModal.tsx:50
 #: components/navigation/Tabs/GroupSettingsTab.tsx:66
-msgid "Group Description"
-msgstr "Gruppenbeschreibung"
+msgid "Group description"
+msgstr "Gruppen Beschreibung"
+
+#: components/modal/CreateGroupModal.tsx:38
+msgid "Group name"
+msgstr "Gruppen Name"
```
Comment (4KevR) [inline frontend/locales/de/messages.po:280]: s.a. to not forget if we change it

**Review by 4KevR — CHANGES_REQUESTED** [review]
Looks great, just small nits

**Review by memeToasty — APPROVED** [review]

File: frontend/components/deck/DeckList.tsx [hunk 5]
```diff
@@ -23,16 +31,23 @@ interface DeckListProps {
...
 	header,
 	decks,
+	filter = "",
+	reverse = false,
 	className = "",
 }: Readonly<DeckListProps>) {
+	const filteredDecks = decks?.filter((deck) =>
+		deck.deckName.includes(filter)
+	);
+	const sortedDecks = reverse ? filteredDecks?.reverse() : filteredDecks;
```
Comment (4KevR) [inline frontend/components/deck/DeckList.tsx:41]: `.reverse` lead to a SonarCloud issue. Maybe use `toReversed()` (non-mutating) instead.

File: frontend/components/group/GroupList.tsx [hunk 6]
```diff
@@ -16,11 +24,18 @@ interface GroupListProps {
...
 export default function GroupList({
 	groups,
+	filter = "",
+	reverse = false,
 	className,
 }: Readonly<GroupListProps>) {
+	const filteredGroups = groups?.filter(
+		(group) => !group.isDefault && group.groupName.includes(filter)
+	);
+	const sortedGroups = reverse ? filteredGroups?.reverse() : filteredGroups;
```
Comment (4KevR) [inline frontend/components/group/GroupList.tsx:34]: s.a.

And another idea. I am not a big React expert but could it make sense to use a `useMemo` here to wrap the filter and reverse operations?

**Review by 4KevR — COMMENTED** [review]
I like the change! I have 2 comments for clarification.

File: frontend/components/modal/CreateGroupModal.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+						placeholder={_(msg`Enter group name`)}
+						className="bg-gray-100 px-2 py-3"
+						inputFieldSize="5xs"
+						autoFocus
+						ref={groupNameInput}
+					></InputField>
+					<InputField
+						id="groupDescriptionInputFieldId"
+						type="text"
+						label={_(msg`Group description`)}
```
Comment (TomRomeo) [inline frontend/components/modal/CreateGroupModal.tsx:50]:
```suggestion
						label={_(msg`Group Description`)}
```
I think we should use Title Case here, as we do with the group name as well.

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx [hunk 8]
```diff
@@ -63,7 +63,7 @@ export const GroupSettingsTab = ({
 				<hr className="border-kiokuLightBlue" />
 				<InputAction
 					id="GroupDescriptionInputAction"
-					header={_(msg`Group Description`)}
+					header={_(msg`Group description`)}
```
Comment (TomRomeo) [inline frontend/components/navigation/Tabs/GroupSettingsTab.tsx:66]: If we decide to use Title Case as suggested above, we should change it here too.

**Review by TomRomeo — APPROVED** [review]

File: frontend/components/deck/DeckList.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/components/deck/DeckList.tsx]: Looks like `toReversed` is the more efficient solution to reverse an array.

File: frontend/components/group/GroupList.tsx (hunk 6 above)
Comment (Julian702) [reply frontend/components/group/GroupList.tsx]: Yes, you are absolutely right. Thanks!

File: frontend/components/modal/CreateGroupModal.tsx (hunk 7 above)
Comment (Julian702) [reply frontend/components/modal/CreateGroupModal.tsx]: fixed in a39d1014ca5d856c1da2ee500c221e7f3503f244

File: frontend/components/deck/DeckList.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/components/deck/DeckList.tsx]: fixed in a39d1014ca5d856c1da2ee500c221e7f3503f244

File: frontend/components/navigation/Tabs/GroupSettingsTab.tsx (hunk 8 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/GroupSettingsTab.tsx]: fixed in a39d1014ca5d856c1da2ee500c221e7f3503f244

File: frontend/components/deck/DeckList.tsx (hunk 5 above)
Comment (4KevR) [reply frontend/components/deck/DeckList.tsx]: `toReversed` is new in ES2023. Our target definition of TS (tsconfig.json) is still on ES5 and our TS version does not include the ES2023 features via esnext in the lib definition. I think we would have to update to at least TS 5.2 to get the new functions. I think that's why the build is not working, but it is interesting that dev still works.

Comment (Julian702) [reply frontend/components/deck/DeckList.tsx]: Ahh, thank for the tip! Now it's working :)

**Review by 4KevR — APPROVED** [review]

**Pull request merged**